lib/auth/trustedcluster_test.go

```diff
@@ -391,6 +393,30 @@ func TestValidateTrustedCluster(t *testing.T) {
 			[]types.CertAuthType{resp.CAs[0].GetType(), resp.CAs[1].GetType(), resp.CAs[2].GetType()},
 		)
 	})
+
+	t.Run("trusted clusters prevented on cloud", func(t *testing.T) {
+		beforeModules := modules.GetModules()
+		modules.SetModules(&modules.TestModules{
+			TestFeatures: modules.Features{Cloud: true},
+		})
+		t.Cleanup(func() { modules.SetModules(beforeModules) })
+
+		req := &ValidateTrustedClusterRequest{
+			Token: "invalidtoken",
+			CAs:   []types.CertAuthority{},
+		}
+
+		server := ServerWithRoles{authServer: a}
+		_, err := server.ValidateTrustedCluster(ctx, req)
+		require.ErrorIs(t, err, trace.NotImplemented("cloud clusters do not support trusted cluster resources"))
+
+		modules.SetModules(&modules.TestModules{
+			TestFeatures: modules.Features{Cloud: false},
+		})
+
+		_, err = server.ValidateTrustedCluster(ctx, req)
+		require.ErrorIs(t, err, trace.AccessDenied("the remote server denied access: invalid cluster token"))
```

Comment:
Same here. I'd change it to assert as well, to cultivate the habit of keeping going.